app/Jobs/EDocument/CreateEDocument.php

```diff
@@ -0,0 +1,139 @@
+                case "XInvoice_2_2":
+                case "XInvoice_2_1":
+                case "XInvoice_2_0":
+                case "XInvoice_1_0":
+                case "XInvoice-Extended":
+                case "XInvoice-BasicWL":
+                case "XInvoice-Basic":
+                    $zugferd = (new ZugferdEDokument($this->document))->run();
+
+                    return $this->returnObject ? $zugferd->xdocument : $zugferd->getXml();
+                case "Facturae_3.2":
+                case "Facturae_3.2.1":
+                case "Facturae_3.2.2":
+                    return (new FacturaEInvoice($this->document, str_replace("Facturae_", "", $e_document_type)))->run();
+                default:
+
+                    $zugferd = (new ZugferdEDokument($this->document))->run();
+
+                    return $this->returnObject ? $zugferd : $zugferd->getXml();
+
+            }
+        }
+        elseif ($this->document instanceof Quote){
+            switch ($e_quote_type){
+                case "OrderX_Basic":
+                case "OrderX_Comfort":
+                case "OrderX_Extended":
+                    $orderx = (new OrderXDocument($this->document))->run();
+                    return $this->returnObject ? $orderx->orderxdocument : $orderx->getXml();
+                default:
+                    $orderx = (new OrderXDocument($this->document))->run();
+                    return $this->returnObject ? $orderx->orderxdocument : $orderx->getXml();
+            }
+        }
+        elseif ($this->document instanceof PurchaseOrder){
+            switch ($e_quote_type){
+                case "OrderX_Basic":
+                case "OrderX_Comfort":
+                case "OrderX_Extended":
+                    $orderx = (new OrderXDocument($this->document))->run();
+                    return $this->returnObject ? $orderx->orderxdocument : $orderx->getXml();
+                default:
+                    $orderx = (new OrderXDocument($this->document))->run();
+                    return $this->returnObject ? $orderx->orderxdocument : $orderx->getXml();
+            }
+        }
+        elseif ($this->document instanceof Credit) {
+            switch ($e_document_type) {
+                case "EN16931":
+                case "XInvoice_3_0":
+                case "XInvoice_2_3":
+                case "XInvoice_2_2":
+                case "XInvoice_2_1":
+                case "XInvoice_2_0":
+                case "XInvoice_1_0":
+                case "XInvoice-Extended":
+                case "XInvoice-BasicWL":
+                case "XInvoice-Basic":
+                    $zugferd = (new ZugferdEDokument($this->document))->run();
+                    return $this->returnObject ? $zugferd->xdocument : $zugferd->getXml();
+                default:
+                    $zugferd = (new ZugferdEDokument($this->document))->run();
+                    return $this->returnObject ? $zugferd : $zugferd->getXml();
+            }
+        }
```

Comment:
same here, can we consolidate credits with invoices, or is there a specific case we need to handle?

Reply:
Same here as above